static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-root-folder.js

```diff
@@ -2752,9 +2751,37 @@ treeNode.getHtml = function() {
                             var path = (this.activeNode.data.uri);
 
                             var editCb = {
-                                success: function() {
-                                    eventNS.typeAction = "";
-                                    eventNS.oldPath = null;
+                                success: function(contentTO, editorId, name, value, draft) {
+
+                                    eventNS.oldPath = oCurrentTextNode.data.path;
+                                    var pageParameter = CStudioAuthoring.Utils.getQueryParameterURL("page");
+                                    if(oCurrentTextNode.data.browserUri != contentTO.item.browserUri){
+                                        var oCurrentTextNodeOldPath = oCurrentTextNode.data.browserUri;
+                                        oCurrentTextNode.data.browserUri = contentTO.item.browserUri;
+                                        oCurrentTextNode.data.path = contentTO.item.path;
+                                        oCurrentTextNode.data.uri = contentTO.item.uri;
+                                        if(oCurrentTextNodeOldPath == pageParameter){
+                                            var currentURL = CStudioAuthoring.Utils.replaceQueryParameterURL(window.location.href, "page", oCurrentTextNode.data.browserUri);
+                                            window.location.href = currentURL;
+                                        }
+                                    }
+                                    if(CStudioAuthoringContext.isPreview){
+                                        try{
+                                            CStudioAuthoring.Operations.refreshPreview();
```

Comment:
Why are we using this method of reloading?  
Don't we have a cleaner event based way to reload?

I dont want any ALERT statements in the code or window.location stuff.... this needs to be encapsulated.